### ISSUE TITLE
Unarchive Fails To List Files In TGZ From FAT FS

### ISSUE THREAD

**Issue by DaytonG** [ansible/ansible#16078]
##### ISSUE TYPE

- Bug Report
##### ANSIBLE VERSION

```
vagrant@vagrant-ubuntu-trusty-64:/vagrant$ ansible --version
ansible 2.1.0.0
  config file = environments/devlocal/ansible.cfg
  configured module search path = Default w/o overrides
```
##### OS / ENVIRONMENT

Ubuntu 14.04 (vagrant VM)
##### SUMMARY

When upgrading from 2.0.2 to 2.1, I notice that the same VM now fails to use the unarchive module. I'm given the following error:
`==> default: fatal: [127.0.0.1]: FAILED! => {"changed": false, "failed": true, "msg": "Failed to find handler for \"/tmp/apache-activemq-5.10.0-bin.tar.gz\". Make sure the required command to extract the file is installed."}`

But with some digging (and tweaking) noticed the actual error is:
`__main__.UnarchiveError: ('Unable to list files in the archive %s: %s', ('/tmp/apache-activemq-5.10.0-bin.tar.gz', 'invalid header'))`

The tar unarchives properly with `tar xvfz`, and files list properly with `tar -tvf`

I also noticed that downloading the file first, then using unarchive yields identical results. So I don't think the download portion is the issue.

In fact, digging through the commits, this jumps out at me: https://github.com/ansible/ansible-modules-core/commit/82aeaed45d5a283a15e8f62dd641b67d04e9876e#diff-c8620ce6bed12fc5110958c7607f1b03

The commit swaps from using a run_command to using a native python file object.
##### STEPS TO REPRODUCE

```
tasks:
  - name: download and unarchive tar
    unarchive: src=http://archive.apache.org/dist/activemq/5.10.0/apache-activemq-5.10.0-bin.tar.gz dest=/tmp copy=no
```

##### EXPECTED RESULTS

File should be downloaded an unarchived.
##### ACTUAL RESULTS

The file downloads, but then fails to get unarchived.

```
==> default: TASK [activemq : download and unarchive tar] ***********************************
==> default: task path: /vagrant/roles/activemq/tasks/install.yml:14
==> default: fatal: [127.0.0.1]: FAILED! => {"changed": false, "failed": true, "msg": "Failed to find handler for \"/tmp/apache-activemq-5.10.0-bin.tar.gz\". Make sure the required command to extract the file is installed."}
==> default:    to retry, use: --limit @site.retry
```

**Comment by DaytonG** [comment]
I'm betting it's because the file was created on a windows machine, and the python tarfile implementation has some issues with that.

`vagrant@vagrant-ubuntu-trusty-64:~$ file /tmp/apache-activemq-5.10.0-bin.tar.gz 
/tmp/apache-activemq-5.10.0-bin.tar.gz: gzip compressed data, from FAT filesystem (MS-DOS, OS/2, NT)`

vs a locally made tarball:
`vagrant@vagrant-ubuntu-trusty-64:~$ file mytar.tar.gz 
mytar.tar.gz: gzip compressed data, from Unix, last modified: Wed Jun  1 11:40:01 2016`

Looks discussed in https://bugs.python.org/issue24514. If I apply patch https://hg.python.org/cpython/rev/301d7efac3de/ to my local system (really, just adding .strip() on one line), the archive can get a file list.

For backwards compatibility reasons, could we have unarchive fall back to the original file list check, if the tarfile module fails?

**Comment by piffey** [comment]
Ran into this today on a machine running Python 2.7.9. Upgraded to Python 2.7.11 and it unarchives the tarball properly on the other end despite the FAT header. Looks like we will just need to check Python version and run one list files function or the other.

**Comment by andrewrothstein** [comment]
using centos7, python 2.7.5. this similarly stopped working for me with v2.1. about to downgrade.

**Comment by dagwieers** [comment]
We are not (no longer) using tarfile in the unarchive module. That was an unfortunate change in May (https://github.com/ansible/ansible-modules-core/commit/af9855357e851ecbe1e8763622a955fb71d96112) that was reverted (https://github.com/ansible/ansible-modules-core/commit/bf670bfdc2c193345a0365bd84e955005b432f31) early July that got shipped as part ov v2.1.0.0. So I don't think this issue is longer valid.

I am going to close this as resolved_by_pr ansible/ansible#3767

@gundalow Can you close this one ?

**Comment by gundalow** [comment]
Closing this as requested by @dagwieers
